### PR TITLE
Unit tests for Rpkt and Udp

### PR DESCRIPTION
**Pull request by andviane**
Here is the new version with (hopefully) all review findings by Scrye applied:

* SoMsg is now used everywhere with the reason explaining the test failure
* The number of blank lines is now much reduced.
* Basically everything is now moved into Convey
* ShouldResemble is consistently used everywhere.
* Underscores are removed from names.
* Cache test is removed from this pull request.

---
This change is [Reviewable](https://reviewable.io/reviews/scionproto/scion/1355)

### CODE REVIEW

**Comment by scrye** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[go/border/rpkt/rpkt_hook_test.go, line 66 at r1](https://reviewable.io:443/reviews/scionproto/scion/1355#-L0ylPKP_E5XB5_np0Dj:-L0ylPKQREzhE5rDGnDp:b-o29nt5) ([raw file](https://github.com/scionproto/scion/blob/b53b494e96cd6bffcc95f1aa8de62e25d5484060/go/border/rpkt/rpkt_hook_test.go#L66)):*
> ```Go
> 		ia, err = rpkt.DstIA()
> 		SoMsg("Destination address wrong on second getter call", ia, ShouldEqual, dstIA)
> 		SoMsg("Destination address must only be fetched once, cached value must be reused on the second call",
> ```

We tend to wrap at > 100 lines (unless readability is severely affected). This line (and a few others) can probably be wrapped.

---

*[go/border/rpkt/rpkt_hook_test.go, line 208 at r1](https://reviewable.io:443/reviews/scionproto/scion/1355#-L0ygwAhNCrpxerOcVzo:-L0ygwAiBMsc-1lO4Wy0:b6z29q6) ([raw file](https://github.com/scionproto/scion/blob/b53b494e96cd6bffcc95f1aa8de62e25d5484060/go/border/rpkt/rpkt_hook_test.go#L208)):*
> ```Go
> 		rpkt.Release()
> 		SoMsg("Release() must decrement ref count from 3 to 2", rpkt.refCnt, ShouldEqual, 2)
> 		SoMsg("Incorrectly freed, 2 refs remaining", free, ShouldBeFalse)
> ```

The message here is a bit confusing; maybe "Not freed yet, 2 refs remaining"?

---

*[go/border/rpkt/rpkt_hook_test.go, line 211 at r1](https://reviewable.io:443/reviews/scionproto/scion/1355#-L0yh2FUTk3cxDQpJk5u:-L0yh2FUTk3cxDQpJk5v:b2nreh6) ([raw file](https://github.com/scionproto/scion/blob/b53b494e96cd6bffcc95f1aa8de62e25d5484060/go/border/rpkt/rpkt_hook_test.go#L211)):*
> ```Go
> 
> 		rpkt.Release()
> 		SoMsg("Incorrectly freed, 1 ref remaining", free, ShouldBeFalse)
> ```

Same as above.

---

*[go/border/rpkt/rpkt_hook_test.go, line 215 at r1](https://reviewable.io:443/reviews/scionproto/scion/1355#-L0yh5W2TlRzJtc_ax8p:-L0yh5W2TlRzJtc_ax8q:bso1ctl) ([raw file](https://github.com/scionproto/scion/blob/b53b494e96cd6bffcc95f1aa8de62e25d5484060/go/border/rpkt/rpkt_hook_test.go#L215)):*
> ```Go
> 		rpkt.Release()
> 		SoMsg("Should be no more refs remaining by now", rpkt.refCnt, ShouldEqual, 0)
> 		SoMsg("Due no refs remaining by now be freed now", free, ShouldBeTrue)
> ```

Suggest rewording to "Freed due to no refs remaining".

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/scionproto/scion/1355)*

**Comment by andviane** [comment]
I have applied the review findings. Hence SoMsg text must document the expected behavior (not the incorrect behavior that would cause this test to fail). Formatting tool normally cares about line length
but not if the excess length is due too long text literal.

**Comment by scrye** [comment]
Sorry for the very late response. I've taken another look at the PR and everything looks good now. Please do a final rebase on our current `master` branch s.t. we can merge your contribution. Thanks for the help and patience!

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/scionproto/scion/1355#-:-L4agXxvs6jfB42i_kjC:buktf2b)*

**Comment by scrye** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/scionproto/scion/1355#-:-L4agvdpgHvV_99Z4I2H:bnfp4nl)*
